src/dodal/devices/fast_grid_scan.py

```diff
@@ -181,181 +190,176 @@ def non_integer_dwell_time(cls, dwell_time_ms: float) -> float:
+    with X as the fast axis and Y as the slow axis. If Z steps isn't 0, the goniometer will
+    then rotate in the omega direction such that it moves from the X-Y, to the X-Z plane then
+    do a second grid scan. The detector is triggered after every x step.
+    See https://github.com/DiamondLightSource/hyperion/wiki/Coordinate-Systems for more
+    """
 
+    def __init__(self, prefix: str, name: str = "") -> None:
+        super().__init__(name)
+        self.x_steps = epics_signal_rw_rbv(int, "X_NUM_STEPS")
+        self.y_steps = epics_signal_rw_rbv(
+            int, "X_NUM_STEPS"
+        )  # Number of vertical steps during the first grid scan
+        self.z_steps = epics_signal_rw_rbv(
+            int, "X_NUM_STEPS"
+        )  # Number of vertical steps during the second grid scan, after the rotation in omega
+        self.x_step_size = epics_signal_rw_rbv(float, "X_STEP_SIZE")
+        self.y_step_size = epics_signal_rw_rbv(float, "Y_STEP_SIZE")
+        self.z_step_size = epics_signal_rw_rbv(float, "Z_STEP_SIZE")
+        self.x_start = epics_signal_rw_rbv(float, "X_START")
+        self.y1_start = epics_signal_rw_rbv(float, "Y_START")
+        self.y2_start = epics_signal_rw_rbv(float, "Y2_START")
+        self.z1_start = epics_signal_rw_rbv(float, "Z_START")
+        self.z2_start = epics_signal_rw_rbv(float, "Z2_START")
+
+        self.position_counter = epics_signal_rw(
+            int, "POS_COUNTER", write_pv="POS_COUNTER_WRITE"
+        )
+        self.x_counter = epics_signal_r(int, "X_COUNTER")
+        self.y_counter = epics_signal_r(int, "Y_COUNTER")
+        self.scan_invalid = epics_signal_r(float, "SCAN_INVALID")
+
+        self.run_cmd = epics_signal_rw(int, "RUN.PROC")
+        self.stop_cmd = epics_signal_rw(int, "STOP.PROC")
+        self.status = epics_signal_r(float, "SCAN_STATUS")
+
+        self.expected_images = ExpectedImages(parent=self)
+
+        self.motion_program = MotionProgram(prefix)
+
+        # Kickoff timeout in seconds
+        self.KICKOFF_TIMEOUT: float = 5.0
+
+        self.COMPLETE_STATUS: float = 60.0
+
+        self.movable_params: dict[str, Signal] = {
+            "x_steps": self.x_steps,
+            "y_steps": self.y_steps,
+            "z_steps": self.z_steps,
+            "x_step_size": self.x_step_size,
+            "y_step_size": self.y_step_size,
+            "z_step_size": self.z_step_size,
+            "x_start": self.x_start,
+            "y1_start": self.y1_start,
+            "y2_start": self.y2_start,
+            "z1_start": self.z1_start,
+            "z2_start": self.z2_start,
+        }
 
-class FastGridScan(Device):
-    x_steps = Component(EpicsSignalWithRBV, "FGS:X_NUM_STEPS")
-    y_steps = Component(EpicsSignalWithRBV, "FGS:Y_NUM_STEPS")
-    z_steps = Component(EpicsSignalWithRBV, "FGS:Z_NUM_STEPS")
+    def is_invalid(self) -> bool:
+        if "GONP" in self.scan_invalid.source:
```

Comment:
I feel like this could do with some explanation

Reply:
This is cargo-culted in from GDA, I actually think we can probably remove it